share/completions/loginctl.fish

```diff
@@ -30,7 +30,7 @@ complete -c loginctl -f -n "not $seen $commands" -l version -d "Show package ver
 
 
 function __fish_loginctl_list_sessions
-    loginctl list-sessions --no-legend --no-pager --output=short | string replace -r '(\d+) \d+ (\S+) (\S*|\s+) (\S+) .*' '$1\t$2 at $4'
+    loginctl list-sessions --no-legend --no-pager --output=short | string replace -r '(\d+)\s+\d+\s+(\w+)\s+(seat\d+\s+)?(tty\d+|pts\/\d+).*' '$1\t$2 at $4'
```

Comment:
1. Those are actual words you're matching, they might very well be localized. If you did want to match them you would have to pass a locale to `loginctl` - `LC_ALL=C loginctl ...`.
2. `\w` ("word"-characters) is not a very useful class to use here. This output will have useful information *every line* (and if it didn't you would have to throw it away). It should be `\S` ("not whitespace").
3. What if there is something that's not "tty" or "pts"?

In essence this runs the risk of under-matching (i.e. not matching input that should be matched), and hence would let untransformed input through.

I like the earlier version more than this.

Reply:
Got that.

Reply:
Columns are aligned with spaces, this regex looks better.